### DISCUSSION TITLE
Using MongoDB instrumentation library with Manual Opentelemetry Config Instrumentation

### DISCUSSION THREAD

**Discussion by indhuvisu** [open-telemetry/opentelemetry-java-instrumentation#12097] [Q&A]
Hi,
I have spring boot project with Opentelemetry instrumentation configured maually. In our project currently we are using mongodb 4.X. Im trying to understand can we use the existing mongdb instrumentation library to capture the query details on spans. I have tried to use **io.opentelemetry.javaagent.instrumentation:opentelemetry-javaagent-mongo-4.0**  in my project. But I don't see any spans created on my command execution. I suspect these instrumentation libraries only will work with javaagent.

**Comment by laurit** [comment]
What you need is https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/mongo/mongo-3.1/library The ones with `javaagent` in the name are for the java agent and can not be used for manual instrumentation.
